three-cards-divination/src/test/java/com/cyber/mysticism/tarot/service/ThreeCardsDivinationServiceTest.java

```diff
@@ -1,63 +1,114 @@
 package com.cyber.mysticism.tarot.service;
 
 import com.cyber.mysticism.tarot.json.Card;
+import com.cyber.mysticism.tarot.model.TarotUser;
 import com.cyber.mysticism.tarot.repository.TarotDeckRepository;
+import com.cyber.mysticism.tarot.repository.ThreeCardsDivinationRepository;
+import com.cyber.mysticism.tarot.repository.UserRepository;
 import com.cyber.mysticism.tarot.service.exceptions.DivinationException;
+import com.cyber.mysticism.tarot.service.exceptions.UserNotFoundException;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.boot.test.mock.mockito.MockBean;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
 
 import java.util.List;
 import java.util.Map;
+import java.util.Optional;
 
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertAll;
 import static org.junit.jupiter.api.Assertions.assertThrows;
-import static org.mockito.Mockito.when;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
 
-@SpringBootTest
+@ExtendWith(MockitoExtension.class)
 class ThreeCardsDivinationServiceTest {
 
-    @Autowired
-    private ThreeCardsDivinationService service;
-    @MockBean
+    @Mock
     private TarotDeckRepository tarotDeckRepository;
+    @Mock
+    private UserRepository userRepository;
+    @Mock
+    private ThreeCardsDivinationRepository threeCardsDivinationRepository;
+    @Mock
+    private TarotUser tarotUser;
+    @InjectMocks
+    private ThreeCardsDivinationService service;
 
-    @Test
-    void getReading() throws Exception {
-        List<Card> cards = List.of(new Card("The Fool", 0, "Major Arcana", "0", List.of()),
+    private static List<Card> getCards() {
+        return List.of(new Card("The Fool", 0, "Major Arcana", "0", List.of()),
                 new Card("The Magician", 1, "Major Arcana", "1", List.of()),
                 new Card("Seven of Cups", 7, "Minor Arcana", "7", List.of("You're being fed a line")));
```

Comment:
**Suggestion:** Consider using `@BeforeEach` to setup common test data and reduce code duplication. [best practice]
```suggestion
    private List<Card> cards;

    @BeforeEach
    void setup() {
        cards = List.of(new Card("The Fool", 0, "Major Arcana", "0", List.of()),
                new Card("The Magician", 1, "Major Arcana", "1", List.of()),
                new Card("Seven of Cups", 7, "Minor Arcana", "7", List.of("You're being fed a line")));
    }
```